libpromises/cf3lex.l

```diff
@@ -133,6 +138,57 @@ promise_type   [a-zA-Z_]+:
                           P.line_pos = 0;
                       }
 
+{macro_if}            {
+                        const char* version_text = yytext+18;
+                        ParserDebug("\tL:macro @if %d:version=%s\n", P.line_pos, version_text);
+                        {
+                          int request_major = 0;
+                          int request_minor = 0;
+                          int major = 0;
+                          int minor = 0;
+                          int patch = 0;
+                          if (3 == sscanf(Version(), "%d.%d.%d", &major, &minor, &patch))
+                          {
+                            if (2 == sscanf(version_text, "%d.%d", &request_major, &request_minor))
+                            {
+                              if (request_major > major || request_minor > minor)
+                              {
+                                /* ignore to the next @endif */
+                                ParserDebug("\tL:macro @if %d:ignoring to next @endif or EOF\n", P.line_pos);
+                                BEGIN(if_state);
+                              }
+                              else /* we're OK with the version */
+                              {
+                                ParserDebug("\tL:macro @if %d:accepted to next @endif or EOF\n", P.line_pos);
+                              }
+                            }
+                            else
+                            {
+                              yyerror("fatal: macro @if requested an unparseable version");
+                              exit(1);
```

Comment:
Sorry, I wasn't being very clear there, what I meant was that we shouldn't be exiting at all, but return an error to the calling function. There are no other code paths in ``cf3lex.l`` that call exit, and the best thing is to let the caller handle it.

For example, if you put invalid policy in ``/var/cfengine/inputs``, cf-serverd might try to load it, fail, and keep running with the old policy. But with this it would exit instead.